app/lib/features/bluetooth/backend/bluetooth_low_energy/ble_device.dart

```diff
@@ -0,0 +1,138 @@
+part of 'ble_manager.dart';
+
+/// BluetoothDevice implementation for the 'bluetooth_low_energy' package
+final class BluetoothLowEnergyDevice
+  extends BluetoothDevice<BluetoothLowEnergyManager, BluetoothLowEnergyService, BluetoothLowEnergyCharacteristic, DiscoveredEventArgs> {
+  /// constructor
+  BluetoothLowEnergyDevice(super._manager, super._source);
+
+  @override
+  String get deviceId => source.peripheral.uuid.toString();
+
+  @override
+  String get name => source.advertisement.name ?? deviceId;
+
+  CentralManager get _cm => manager.backend;
+
+  @override
+  Stream<BluetoothConnectionState> get connectionStream => _cm.connectionStateChanged.transform(
+    BluetoothConnectionStateStreamTransformer(stateParser: BluetoothLowEnergyConnectionStateParser())
+  );
+
+  @override
+  Future<void> backendConnect() => _cm.connect(source.peripheral);
+
+  @override
+  Future<void> backendDisconnect() => _cm.disconnect(source.peripheral);
+
+  @override
+  Future<List<BluetoothLowEnergyService>?> discoverServices() async {
+    if (!isConnected) {
+      logger.finer('discoverServices: device not connect. Call device.connect() first');
+      return null;
+    }
+
+    // Query actual services supported by the device. While they must be
+    // rediscovered when a disconnect happens, this object is also recreated.
+    try {
+      final allServices = await _cm.discoverGATT(source.peripheral);
+      logger.finer('discoverServices: $allServices');
+
+      return allServices.map(BluetoothLowEnergyService.fromSource).toList();
+    } catch (e) {
+      logger.shout('discoverServices: error:', [source.peripheral, e]);
+      return null;
+    }
+  }
+  
+  @override
+  Future<bool> getCharacteristicValueByUuid(BluetoothLowEnergyCharacteristic characteristic, List<Uint8List> value) async {
+    if (!isConnected) {
+      logger.finer('getCharacteristicValueByUuid: device not connect. Call device.connect() first');
+      return false;
+    }
+
+    if (characteristic.canRead) { // Read characteristic value if supported
+      try {
+        final data = await _cm.readCharacteristic(
+          source.peripheral,
+          characteristic.source,
+        );
+
+        value.add(data);
+        return true;
+      } catch (err) {
+        return false;
+      }
+    }
+
+    if (characteristic.canIndicate) { // Listen for characteristic value and trigger the device to send it
```

Comment:
notifications and indications are very similar, I read somewhere that they are often abstracted away.
But when I implemented ble, using the wrong one resulted in not reading the data successfully. So it would be best if you could test that on your device.